src/rust/rust-proto/src/graplinc/grapl/api/event_source/v1beta1/client.rs

```diff
@@ -1,102 +1,83 @@
-use std::time::Duration;
-
-use client_executor::{
-    Executor,
-    ExecutorConfig,
-};
+use tonic::transport::Endpoint;
 
 use crate::{
-    client_factory::services::EventSourceClientConfig,
-    client_macros::RpcConfig,
-    create_proto_client,
-    execute_client_rpc,
-    graplinc::grapl::api::event_source::v1beta1 as native,
-    protobufs::graplinc::grapl::api::event_source::v1beta1::{
-        self as proto,
-        event_source_service_client::EventSourceServiceClient as EventSourceServiceClientProto,
-    },
-    protocol::{
-        endpoint::Endpoint,
-        error::GrpcClientError,
-        service_client::{
-            ConnectError,
+    graplinc::grapl::api::{
+        client::{
+            client_impl,
+            Client,
+            ClientError,
             Connectable,
         },
+        event_source::v1beta1 as native,
     },
+    protobufs::graplinc::grapl::api::event_source::v1beta1::event_source_service_client::EventSourceServiceClient,
 };
 
-pub type EventSourceServiceClientError = GrpcClientError;
+#[async_trait::async_trait]
+impl Connectable for EventSourceServiceClient<tonic::transport::Channel> {
+    async fn connect(endpoint: Endpoint) -> Result<Self, ClientError> {
+        Ok(Self::connect(endpoint).await?)
+    }
+}
```

Comment:
I guess I'm a tad confused here - I assume this isn't a recursive call - is there another `connect` and we're just converting the error type?

Reply:
I'll try to document these a lot more clearly now that I've finished removing the healthcheck stuff. There is a little complexity here due to the desire to maintain encapsulation of the protocol buffer types.

The general idea is that we have a _blanket implementation_ of `Connect<C>` which is our public inteface--you give it a `ClientConfiguration` and it gives you a connected client ready to make requests. So rust-proto users need only import `Connect` and their client of choice--e.g. `PipelineIngressClient`. This will give them the ability to invoke `PipelineIngressClient::connect(client_configuration).await?`.

Reply:
`Self` has type `EventSourceServiceClient<tonic::transport::Channel>` so this is calling the `connect` method on the protocol buffer compiler generated type `EventSourceServiceClient`. The `Connectable` interface is our bridge between the "generated" world and our application types. We implement `Connectable` on the generated types, and then `Connect<C: Connectable>` for our application types.

This sidesteps the issue that the generated types don't provide some interface. No matter, we'll just impose our own interface on them. Aren't traits cool?

Reply:
I hope these comments kind of make it more clear, I'll work on documenting these types now.

Reply:
Docs are done, I've elected to keep the terse type parameters but document what they mean.